src/qibo/models/circuit.py

```diff
@@ -1210,6 +1210,22 @@ def from_qasm(cls, qasm_code, accelerators=None, density_matrix=False):
         parser = QASMParser()
         return parser.to_circuit(qasm_code, accelerators, density_matrix)
 
+    def to_qir(self):
+        """
+        Convert circuit to QIR circuit.
+
+        Uses `qbraid` (https://github.com/qBraid/qBraid) to transpile
+        the circuit into `pyqir` circuits.
+        """
+        try:
+            from qbraid.transpiler.conversions.qasm2 import qasm2_to_qasm3
+            from qbraid.transpiler.conversions.qasm3 import qasm3_to_pyqir
+        except ModuleNotFoundError as e:  # pragma: no cover
+            raise ModuleNotFoundError(
+                "The optional dependency qbraid is missing, please install it with `poetry install --with qir`"
+            ) from e
```

Comment:
As you did for cudaq, you have to import `import qbraid_qir` otherwise the the exception will not tell the missing package.